pages/details/[id].js

```diff
@@ -24,6 +24,21 @@ export default function Details() {
   const destinationName = destination?.name || "Not found";
   const destinationQueryName = destinationName.replaceAll(" ", "-");
 
+  const onDeleteToDoItem = (deleteId) => {
```

Comment:
Nice, but you can try to get into a method of writing your functions with expressions instead of statements:
```
const onDeleteToDoItem = id =>
  setDestinations(destinations =>
    destinations.map(destinationItem =>
      destinationItem.id === destination.id
        ? {
            ...destinationItem,
            toDos: destinationItem.toDos.filter(toDo => toDo.id !== id),
          }
        : destinationItem,
    ),
  );
```

Reply:
Done.